spec/classes/stub_spec.rb

```diff
@@ -0,0 +1,275 @@
+require 'spec_helper'
+
+describe 'bin/stub' do
+  let(:subject_path) { File.expand_path('stub', "#{File.dirname(__FILE__)}/../../bin") }
+  let(:stub_call_pathname) { instance_double('Pathname') }
+  let(:stub_directory_pathname) { instance_double('Pathname') }
+  let(:stub_config_pathname) { instance_double('Pathname') }
+  let(:stub_output_string_pathname) { instance_double('Pathname') }
+  let(:stub_call_file) { StringIO.new }
+  let(:stub_config_file) { StringIO.new }
+  let(:stub_stderr_file) { StringIO.new }
+  let(:stub_stdout_file) { StringIO.new }
+  let(:stub_output_string_file) { StringIO.new }
+  let(:exit_code_list) { [] }
+
+  before(:each) do
+    allow(stub_call_pathname).to receive(:open).with('a')
+      .and_yield(stub_call_file)
+    allow(stub_output_string_pathname).to receive(:open).with('w')
+      .and_yield(stub_output_string_file)
+    allow(stub_config_pathname).to receive(:exist?)
+
+    allow(Pathname).to receive(:new).with('a unique file name')
+      .and_return(stub_output_string_pathname)
+    allow(Pathname).to receive(:new).with('first_argumentsecond_argumenta unique file name')
+      .and_return(stub_output_string_pathname)
+
+    allow(Pathname).to receive(:new).with(File.dirname(subject_path))
+      .and_return(stub_directory_pathname)
+
+    allow(stub_directory_pathname).to receive(:join).with('stub_calls.yml')
+      .and_return(stub_call_pathname)
+    allow(stub_directory_pathname).to receive(:join).with('stub_stub.yml')
+      .and_return(stub_config_pathname)
+
+    $stderr = stub_stderr_file
+    $stdout = stub_stdout_file
+    allow($stdin).to receive(:tty?).and_return(false)
+
+    allow_any_instance_of(Kernel).to receive(:exit) do |_, exit_code|
+      exit_code_list << exit_code
+      raise SystemExit
+    end
+  end
+
+  context 'with no configuration' do
+    let(:stub_call_log) do
+      ARGV.replace []
```

Comment:
In hindsight, I think I can break these out into individual tests to make it WETter. This will be slightly repetitive, but will make the actual behavior more clear. I can also clean up the wording on the contexts and tests to make things more clear.